### PR TITLE
Documented when helper requests get queued

### PR DESCRIPTION
**Pull request by rousskov**
I had to change introductory paragraphs in several directives so that
the new documentation can refer to "numberofchildren". I fixed a few
spelling/grammar problems in changed paragraphs and edited them a bit
for consistency, but they need more work.

Motivation:
http://lists.squid-cache.org/pipermail/squid-users/2018-July/018577.html

### CODE REVIEW

**Comment by rousskov** [comment]
This PR should be proof-read by a fresh pair of eyes.

File: src/cf.data.pre [hunk 1]
```diff
@@ -3284,9 +3289,14 @@ IFDEF: USE_SSL_CRTD
 DEFAULT: 32 startup=5 idle=1
 LOC: Ssl::TheConfig.ssl_crtdChildren
 DOC_START
-	The maximum number of processes spawn to service ssl server.
-	The maximum this may be safely set to is 32.
-	
+	Specifies the maximum number of certificate generation processes that
+	Squid may spawn (numberofchildren) and several related options. Using
+	too few helper processes queues requests. Using too many helpers
```
Comment (amishxda) [inline src/cf.data.pre]: It should be called either "helper processes" or just "helpers".

File: src/cf.data.pre [hunk 2]
```diff
@@ -3338,8 +3349,13 @@ IFDEF: USE_OPENSSL
 DEFAULT: 32 startup=5 idle=1 concurrency=1
 LOC: Ssl::TheConfig.ssl_crt_validator_Children
 DOC_START
-	The maximum number of processes spawn to service SSL server.
-	The maximum this may be safely set to is 32.
+	Specifies the maximum number of certificate validation processes that
+	Squid may spawn (numberofchildren) and several related options. Using
+	too few helper processes queues requests. Using too many helpers
```
Comment (amishxda) [inline src/cf.data.pre]: It should be called either "helper processes" or just "helpers".

File: src/cf.data.pre [hunk 3]
```diff
@@ -5462,11 +5479,13 @@ TYPE: HelperChildConfig
...
 LOC: Config.redirectChildren
 DOC_START
-	The maximum number of redirector processes to spawn. If you limit
-	it too few Squid will have to wait for them to process a backlog of
-	URLs, slowing it down. If you allow too many they will use RAM
-	and other system resources noticably.
-	
+	Specifies the maximum number of redirector processes that Squid may
+	spawn (numberofchildren) and several related options. Using too few
+	helper processes queues requests. Using too many helpers wastes RAM
```
Comment (amishxda) [inline src/cf.data.pre]: It should be called either "helper processes" or just "helpers".

File: src/cf.data.pre (hunk 1 above)
Comment (rousskov) [reply src/cf.data.pre]: Please see if you like the rephrased version better.

File: src/cf.data.pre (hunk 2 above)
Comment (rousskov) [reply src/cf.data.pre]: Please see if you like the rephrased version better.

File: src/cf.data.pre (hunk 3 above)
Comment (rousskov) [reply src/cf.data.pre]: Please see if you like the rephrased version better.

**Comment by amishxda** [comment]
Well I am not a documentation or English expert. But yes its better.

I am not sure if intention is to be accurate grammatically but if it is then I would suggest "comma" and "the" like this:

Using too few of these helper processes (a.k.a. "helpers")**_,_** queues **_the_** requests. Using too many helpers**_,_** wastes **_the_** RAM and other system resources.

Actually "system resources" implies RAM so we can shorten the text to "wastes the system resources"

**Comment by rousskov** [comment]
> I am not sure if intention is to be accurate grammatically

Yes, we should write grammatically accurate prose! I do appreciate you taking a critical look at this text.

> I would suggest "comma" 

Separating subject from its verb with a comma is wrong. See example [1a](http://grammar.ccc.commnet.edu/grammar/gerunds.htm) for a more authoritative reference and a more detailed explanation. I also tested the variants with a comma on another person unfamiliar with the text -- the comma confused them quite a bit!

> queues *the* requests

This one is debatable, but I rephrased to avoid the problem. Please see of the current version is better.

> wastes *the* system resources

This one is debatable, but I rephrased to avoid the problem. Please see of the current version is acceptable.

> Actually "system resources" implies RAM so we can shorten the text

Done.

**Comment by yadij** [comment]
@amishxda, everyones input is appreciated.

**Pull request closed without merging**